### DISCUSSION TITLE
限定共有記事の投稿日と編集履歴をリセットして公開したい

### DISCUSSION THREAD

**Discussion by uhooi** [increments/qiita-discussions#25] [要望・アイデア・不具合]
現在、限定共有記事を公開するときは投稿日と編集履歴が残ります。
私は編集履歴は残したいのでこのままでいいのですが、投稿日は公開日に更新したいことがあります。

人によっては編集履歴にプライベートな内容が残っている可能性があるので、その場合はわざわざ記事の内容ををコピペして新規投稿しなければならず、手間となってしまいます。

**Comment by uhooi** [comment]
私が投稿日を更新したい理由は、自分の記事一覧に最新として表示してほしいためです。
しかし今見てみたら、投稿日は更新されなくても記事一覧には公開日が反映されているようでした。おそらく内部的に公開日を持たせていて、それをソートに使っているようです。

**Reply by uhooi** [reply]
（前は投稿日がソートに使われていた気がする…）

**Reply by tanaka-asami415** [reply]
@uhooi 
「限定共有記事を全体公開した際に、投稿日を公開日とする」対応について、2023年に完了しておりましたが、ご案内ができておりませんでした🙇
ご報告が遅くなってしまい申し訳ありません。

**Reply by uhooi** [reply]
なんと！笑
ご対応ありがとうございます :sparkles:

**Comment by kskwtnk** [comment]
@uhooi 

ご意見ありがとうございます！
とても分かります、記事一覧のうち結構下の方に記事が挿入されてびっくりした経験あります……

ただ、若干の難しさは感じていました

- 編集履歴を残す残さないをカスタムするのは難しい
- https://github.com/qiita-discussions/qiita-discussions/discussions/15 こちらの要望もよく挙がるため、公開→非公開→公開……と繰り返した際、日付や編集履歴をどういう扱いにするか

こういった理由もあり、どういう仕様として実現するか悩むポイントです

---

履歴や公開日など、他のみなさんからのご意見もお待ちしています！

**Reply by daikisuyama** [reply]
返信失礼します。
「公開→非公開→公開……」と繰り返すパターンはどういった場合を想定しているかを教えていただきたいです。
また、投稿日を公開日に更新する方がメリットが大きいと個人的には思うのですが、どういったデメリットがあるかも教えていただきたいです。

**Reply by kskwtnk** [reply]
@daikisuyama 

ありがとうございます！

> 「公開→非公開→公開……」と繰り返すパターンはどういった場合を想定しているかを教えていただきたいです。

例えばなのですが

- 記事が炎上してしまって一旦非公開にした、しかしほとぼりが冷めたのでまた公開したい……
- 非公開→公開した記事の内容が間違っていて「会社所属を表明したまま間違った記事を出すのはNG」と言われ一旦非公開、また修正して公開

などといったケースを想定していました
1件目の炎上の件は、それに近いお問い合わせをいただいたこともあり……結局記事を削除して終わりになってしまったのですが、もし非公開と公開を自由に選べた場合は恐らくこういう対応をしていたのだろうと思われます

投稿日と更新日の話は、データの持ち方として「今はそれに適した形式になっていない」が理由として大きいため、デメリットは我々としてもそこまで認識していません
どちらかと言えばネックになるのは「実装としてどうしようか……」の話だと思います 😢

**Reply by daikisuyama** [reply]
公開/非公開を選べる場合、公開時にブックマークしていたユーザが非公開時にもアクセスできるのかや何度も公開/非公開することを良しとするのかなど色々な問題が発生するのかなと感じました。また、投稿日と更新日についても承知しました。

仕様の策定や実装など、難しい点が沢山おありかと存じますが、応援しております。

**Comment by blono** [comment]
横からですが、

1. 公開する
2. 内容に対して色々誤りが指摘される
3. 一旦非公開とし、修正する
4. 修正が終わり次第再度公開する

があるかなって思いました！

**Reply by daikisuyama** [reply]
説明が不足していて申し訳ないです。
@blonoさんにご紹介いただいたパターンは想定していたのですが、4での再度公開後に公開/非公開を繰り返すパターンが自分には想定できませんでした。
また、4での再度公開後に公開/非公開を繰り返した場合、ブックマーク等をしていてもURLが変わってその記事を閲覧できなくなると思うので、なんらかの対策が必要であるとも思います。

**Comment by blono** [comment]
こちらこそ読み込めておらず申し訳ないです。

URL変わってしまうんですね、知りませんでした。
現状、公開->非公開が想定・実装されていない機能になるので、そういう作りなのでしょうね。

別問題として、URLは一つの記事につきどんな操作をしても固定であるべきではないかと感じました。
例えば、ユーザー名を変更すると記事のURLが全部変わりますが、果たしてそれでいいのか、とかちょっと疑問に思いました！🤔

**Reply by kskwtnk** [reply]
@blono 

> 例えば、ユーザー名を変更すると記事のURLが全部変わりますが、果たしてそれでいいのか、とかちょっと疑問に思いました！🤔

こちら補足なのですが、ユーザー名が変わるとURLは変わるものの記事のidは変わらないため、ブックマークからは問題なく飛ぶことができます
若干話している内容とは違うと思いますが、URLが変わって困るシーンの代表というとブックマークからのアクセスかな思い補足します！

**Comment by k1350** [comment]
Qiitaへの投稿に不慣れで、記事の編集履歴が公開されてしまうことを知りませんでした。
限定共有状態で色々修正してからそのまま公開してしまっていたので、このissueで編集履歴が公開されてしまうことを知り、今まさに新規投稿しなおしたところです。

それほど多くの投稿サイトを使ってはいませんが、そもそも編集履歴が公開されてしまう仕様の投稿サイトをあまり見かけません。

> 人によっては編集履歴にプライベートな内容が残っている可能性がある

という点については、私のような初心者の場合は「公開したくない内容が残っている上に、投稿者本人が気づかずそのまま公開し続けている」という問題が実はあるのではないかと思います。

> 編集履歴を残す残さないをカスタムするのは難しい

とのことですが、「編集履歴をリセットして全体公開する」という選択肢がUI上に存在すれば「そもそも編集履歴が公開されるのだ」ということに気づけると思いますし、わざわざ新規投稿を作り直す必要も無いので、残す・残さないをカスタムできると初心者としてはありがたいです。

**Reply by kskwtnk** [reply]
@k1350 

ご意見ありがとうございます！

会員登録時などでも全く説明がなく、その点については申し訳ありません 🙇 
技術情報の場合はいつ出した情報か（いつアップデートした情報か）もある程度重要だと考えているため、履歴を残すようにしています

> 「編集履歴をリセットして全体公開する」という選択肢がUI上に存在すれば

編集履歴関連については多くご意見をいただいていまして、現状進めているプロジェクトの進捗ともあわせて検討させていただきます！
